sdk/storage/storage-file-datalake/README.md

```diff
@@ -0,0 +1,520 @@
+
+This project provides a client library in JavaScript that makes it easy to consume Microsoft Azure Storage Data Lake service.
+
+Use the client libraries in this package to:
+  - Create/List/Delete File Systems
+  - Create/Read/List/Update/Delete Paths, Directories and Files
+
+[Source code](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-file-datalake) |
+[Package (npm)](https://www.npmjs.com/package/@azure/storage-file-datalake) |
+[API Reference Documentation](https://docs.microsoft.com/javascript/api/@azure/storage-file-datalake) |
+[Product documentation](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction?toc=%2fazure%2fstorage%2fblobs%2ftoc.json) |
+[Samples](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-file-datalake/samples) |
+[Azure Storage Data Lake REST APIs](https://docs.microsoft.com/en-us/rest/api/storageservices/data-lake-storage-gen2)
+
+## Key concepts
+
+DataLake Storage Gen2 was designed to:
+
+- Service multiple petabytes of information while sustaining hundreds of gigabits of throughput
+- Allow you to easily manage massive amounts of data
+
+Key Features of DataLake Storage Gen2 include:
+
+- Hadoop compatible access
+- A super set of POSIX permissions
+- Cost effective in terms of low-cost storage capacity and transactions
+- Optimized driver for big data analytics
+
+A fundamental part of Data Lake Storage Gen2 is the addition of a hierarchical namespace to Blob storage. The hierarchical namespace organizes objects/files into a hierarchy of directories for efficient data access.
+
+In the past, cloud-based analytics had to compromise in areas of performance, management, and security. Data Lake Storage Gen2 addresses each of these aspects in the following ways:
+- Performance is optimized because you do not need to copy or transform data as a prerequisite for analysis. The hierarchical namespace greatly improves the performance of directory management operations, which improves overall job performance.
+- Management is easier because you can organize and manipulate files through directories and subdirectories.
+- Security is enforceable because you can define POSIX permissions on directories or individual files.
+- Cost effectiveness is made possible as Data Lake Storage Gen2 is built on top of the low-cost Azure Blob storage. The additional features further lower the total cost of ownership for running big data analytics on Azure.
+
+Data lake storage offers three types of resources:
+
+- The _storage account_ used via `DataLakeServiceClient`
+- A _file system_ in the storage account used via `DataLakeFileSystemClient`
+- A _path_ in a file system used via `DataLakeDirectoryClient` or `DataLakeFileClient`
+
+|ADLS Gen2 	                | Blob       |
```

Comment:
>ADLS [](start = 1, length = 4)

nit: ADLS might not be an obvious acronym. We should introduce it first (like HNS below)

Reply:
Will do.